sql-plugin/src/main/scala/com/nvidia/spark/rapids/HostAlloc.scala

```diff
@@ -52,17 +54,32 @@ private class HostAlloc(nonPinnedLimit: Long) extends HostMemoryAllocator with L
     }
   }
 
+  private def getHostAllocMetricsLogStr(metrics: GpuTaskMetrics): String = {
+    Option(TaskContext.get()).map({ context =>
```

Comment:
don't need both ( and {, so:
```
Option(TaskContext.get()).map { context => 
...
}
```
would be better.

Why not put the `logDebug` inside of the `foreach`? that way your return type can be `Unit` and you don't need to call `logDebug` after the fact, and you don't need the `getOrElse`

Reply:
The reason is so that when log level is set to less than debug eg. info, none of the code will have to execute at all

Reply:
ah that makes sense, never mind.